### DISCUSSION TITLE
Some Ideas...

### DISCUSSION THREAD

**Discussion by XeniumMods** [konwektor/BlueRetro#22] [Ideas]
Hello Konwektor,

I would like to help you with this project where I can.
I've been making hardware for the OGXbox for a few years now and
have been wanting a wireless version of the OGX360 for sometime.
I have a number of ideas that may or may not be of value to you.

When we last spoke over at OGXbox.com forums you were
discussing using the eject/power buttons for short/long presses.

I had a similar idea for using the buttons for ARGB controller to select between
patterns and colors. While the console was OFF, the buttons would work as intended.
However, when powered ON I wanted short presses to trigger the ARGB controller.
With long presses (eg. 3 sec or more) to power off and eject the DVD tray.
I was going to test the short/long press concept with an ATtiny45 or something.

![Untitled](https://github.com/konwektor/BlueRetro/assets/112789266/093e06a7-db6d-481a-9143-b7c48c0c4fad)

I just did a quick schematic to help illustrate my idea...

While console is off the PNP transistors bridge connection between *B-IN (button) & *B-OUT (mobo)
When the console is on this bridge is broken and we can use short/long presses via ATtiny

PB-OUT = Long press power button to power off console
EB-OUT = Long Press eject button to eject DVDrom
EN = Short press power button...
Boot = Short press eject button...

Just an idea I had that I thought I would share, let me know what you think...

Cheers, NeMesiS

**Comment by konwektor** [comment]
First:
- with your diagram, using PNP transistor - without Vcc there will be no connection beetwen *IN-*OUT
- when we provide some VCC to Base of transistor - current start flowing from Emiter to Collector - so from *OUT to *IN
everything "diversed" as in Your concept.
I would use one input as a "power detection" connected to xbox "PWR OK - I have talk about it exactly tooday with Crossfader
- If system off, signal LO, 
- If system on signal is HI (3,3V)
- depends of state "power detection" signal (LOW or HI) do some things in "this" or "that" way, after inputPB or inputEB is pushed.
 But you will need to have power to attiny all time - xbox 3,3V STBY (max 100mA) in theory could be enough

**Comment by XeniumMods** [comment]
- Correct... System off = LO and System on = HI
- NPN will connect on HI and PNP will connect on LO
- PNP is used to connect IN and OUT while system is LO not requiring ATtiny45 during this time.
- ATtiny45 will not need to be powered all the time, only when system is HI.

Here's a more detailed schematic...

![Untitled](https://github.com/konwektor/BlueRetro/assets/112789266/10728d8c-6ab5-4355-81ee-3e43e1ce64ff)

**Reply by konwektor** [reply]
Fast response :-)
Can we use Discord?

**Reply by XeniumMods** [reply]
Yea, sure... ;)
